packages/embedder/addEmbeddingsMetadataForRetrospectiveDiscussionTopic.ts

```diff
@@ -0,0 +1,108 @@
+import {ExpressionOrFactory, SqlBool, sql} from 'kysely'
+import getRethink from 'parabol-server/database/rethinkDriver'
+import {RDatum} from 'parabol-server/database/stricterR'
+import getKysely from 'parabol-server/postgres/getKysely'
+import {DB} from 'parabol-server/postgres/pg'
+import {Logger} from 'parabol-server/utils/Logger'
+import {EmbedderOptions} from './embedder'
+
+interface DiscussionMeta {
+  id: string
+  teamId: string
+  createdAt: Date
+}
+
+const validateDiscussions = async (discussions: (DiscussionMeta & {meetingId: string})[]) => {
+  const r = await getRethink()
+  if (discussions.length === 0) return discussions
+  // Exclude discussions that belong to an unfinished meeting
+  const meetingIds = [...new Set(discussions.map(({meetingId}) => meetingId))]
+  const endedMeetingIds = await r
+    .table('NewMeeting')
+    .getAll(r.args(meetingIds), {index: 'id'})
+    .filter((row: RDatum) => row('endedAt').default(null).ne(null))('id')
+    .distinct()
+    .run()
+  const endedMeetingIdsSet = new Set(endedMeetingIds)
+  return discussions.filter(({meetingId}) => endedMeetingIdsSet.has(meetingId))
+}
+
+const insertDiscussionsIntoMetadata = async (discussions: DiscussionMeta[]) => {
+  const pg = getKysely()
+  const metadataRows = discussions.map(({id, teamId, createdAt}) => ({
+    refId: id,
+    objectType: 'retrospectiveDiscussionTopic' as const,
+    teamId,
+    // Not techincally updatedAt since discussions are be updated after they get created
+    refUpdatedAt: createdAt
+  }))
+  if (!metadataRows[0]) return
+  await pg
+    .insertInto('EmbeddingsMetadata')
+    .values(metadataRows)
+    .onConflict((oc) => oc.doNothing())
+    .execute()
+  return
+}
+
+export const addEmbeddingsMetadataForRetrospectiveDiscussionTopic = async ({
+  startAt,
+  endAt,
+  meetingId
+}: EmbedderOptions) => {
+  // load up the metadata table will all discussion topics that are a part of meetings ended within the given date range
+  const pg = getKysely()
+  if (meetingId) {
+    const discussions = await pg
+      .selectFrom('Discussion')
+      .select(['id', 'teamId', 'createdAt'])
+      .where('meetingId', '=', meetingId)
+      .execute()
+    await insertDiscussionsIntoMetadata(discussions)
+    return
+  }
+  const PG_MAX_PARAMS = 65535
```

Comment:
+1 

Consider adding a comment on the why behind the BATCH_SIZE, I'd like to understand